src/vs/workbench/api/common/extHostDebugService.ts

```diff
@@ -1017,7 +1017,7 @@ export class ExtHostDebugSession {
 
 	public get api(): vscode.DebugSession {
 		const that = this;
-		return Object.freeze({
+		return this.apiSession = Object.freeze({
```

Comment:
this still reassigns it on every get

```suggestion
		return this.apiSession ??= Object.freeze({
```